app/javascript/Components/__tests__/create_tag_modal.test.jsx

```diff
@@ -21,14 +22,26 @@ describe("CreateTagModal", () => {
     props = {
       isOpen: true,
       onRequestClose: jest.fn().mockImplementation(() => (props.isOpen = false)),
-      grouping_id: 1,
-      course_id: 1,
-      assignment_id: 1,
     };
 
     // Set the app element for React Modal
     Modal.setAppElement("body");
-    component = render(<CreateTagModal {...props} />);
+
+    const contextValue = {
+      result_id: 1,
+      submission_id: 1,
+      assignment_id: 1,
+      grouping_id: 1,
+      course_id: 1,
+      role: "Student",
+      is_reviewer: true,
+    };
+
+    component = render(
```

Comment:
So overall I think this is a reasonable place for a good test helper function. Drawing inspiration from https://testing-library.com/docs/example-react-context/, perhaps something like `renderInResultContext`. It would be nice if that function set defaults for the different context values, so that the user only needs to specify the ones they want to change for a given test.